blocks/settlement.go

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package blocks
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+)
+
+type ancestry struct {
+	parent, lastSettled *Block
+}
+
+var errBlockResettled = errors.New("block re-settled")
+
+// MarkSettled marks the block as having been settled. This function MUST NOT
+// be called more than once.
+//
+// After a call to MarkSettled, future calls to [Block.ParentBlock] and
+// [Block.LastSettled] will return nil.
+func (b *Block) MarkSettled() error {
+	a := b.ancestry.Load()
+	if a == nil {
+		b.log.Error(errBlockResettled.Error())
+		return fmt.Errorf("%w: block height %d", errBlockResettled, b.Height())
+	}
+	if !b.ancestry.CompareAndSwap(a, nil) { // almost certainly means concurrent calls to this method
+		b.log.Fatal("Block ancestry changed during settlement")
+		// We have to return something to keen the compiler happy, even though we
+		// expect the Fatal to be, well, fatal.
+		return errors.New("block ancestry changed during settlement")
```

Comment:
nit: mind moving this out to a global? I know it should never be hit but seems nice to be consistent

Reply:
Done.